controller/src/controller/subsystems/instrument_comm.py

```diff
@@ -584,6 +652,17 @@ async def _process_command_response(self, packet_type: int, response_data: bytes
                 if self._firmware_update_manager is None:
                     raise NotImplementedError("_firmware_update_manager should never be None here")
                 await self._firmware_update_manager.update(command, response_data)
+            case "check_connection_status":
+                prev_command_info["status"] = response_data[0]
+                self._system_in_offline_mode = response_data[0] == ConnectionStatuses.HEADLESS.value
+                self._offline_state_change.set()
```

Comment:
Do we need to set this if we're in online mode? I think this will cause two of each of the online mode tasks to be running with the way that `_manage_online_mode_tasks` is written right now